huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_directory.go

```diff
@@ -0,0 +1,335 @@
+		d.Set("name", utils.PathSearch("name", directory, nil)),
+		d.Set("type", utils.PathSearch("type", directory, nil)),
+		d.Set("description", utils.PathSearch("description", directory, nil)),
+		d.Set("parent_id", utils.PathSearch("parent_id", directory, nil)),
+		d.Set("root_id", utils.PathSearch("root_id", directory, nil)),
+		d.Set("qualified_name", utils.PathSearch("qualified_name", directory, nil)),
+		d.Set("created_at", utils.PathSearch("create_time", directory, nil)),
+		d.Set("updated_at", utils.PathSearch("update_time", directory, nil)),
+		d.Set("created_by", utils.PathSearch("create_by", directory, nil)),
+		d.Set("updated_by", utils.PathSearch("update_by", directory, nil)),
+		d.Set("children", utils.PathSearch(`children[*].name`, directory, make([]interface{}, 0))),
+	)
+	if err := mErr.ErrorOrNil(); err != nil {
+		return diag.Errorf("error setting DataArts Studio directory fields: %s", err)
+	}
+
+	return nil
+}
+
+func resourceStudioDirectoryUpdate(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	//nolint:misspell
+	updateDirectoryHttpUrl := "v2/{project_id}/design/directorys"
+	updateDirectoryProduct := "dataartsv2"
+
+	updateDirectoryClient, err := cfg.NewServiceClient(updateDirectoryProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DataArts Studio V2 Client: %s", err)
+	}
+	updateDirectoryPath := updateDirectoryClient.Endpoint + updateDirectoryHttpUrl
+	updateDirectoryPath = strings.ReplaceAll(updateDirectoryPath, "{project_id}", updateDirectoryClient.ProjectID)
+
+	updateDirectoryOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"workspace": d.Get("workspace_id").(string)},
+	}
+
+	updateDirectoryOpt.JSONBody = utils.RemoveNil(buildUpdateDirectoryBodyParams(d))
+	updateDirectoryResp, err := updateDirectoryClient.Request("PUT", updateDirectoryPath, &updateDirectoryOpt)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	updateDirectoryRespBody, err := utils.FlattenResponse(updateDirectoryResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	directory := utils.PathSearch("data.value", updateDirectoryRespBody, nil)
+
+	id, err := jmespath.Search("id", directory)
+	if err != nil {
+		return diag.Errorf("error updating DataArts Studio directory: %s is not found in API response", id)
+	}
+
+	qualifiedName, err := jmespath.Search("qualified_name", directory)
+	if err != nil {
+		return diag.Errorf("error updating DataArts Studio directory: %s is not found in API response", qualifiedName)
+	}
+	if qualifiedName == nil {
+		qualifiedName = d.Get("qualified_name")
+	}
+
+	rootID, err := jmespath.Search("root_id", directory)
+	if err != nil {
+		return diag.Errorf("error updating DataArts Studio directory: %s is not found in API response", rootID)
+	}
+
+	d.SetId(id.(string))
+	d.Set("root_id", rootID)
+	d.Set("qualified_name", qualifiedName)
+
+	return nil
```

Comment:
it's better to call `resourceStudioDirectoryRead` at the end of Update function.